src/libraries/System.Formats.Tar/src/System/Formats/Tar/SeekableSubReadStream.cs

```diff
@@ -0,0 +1,111 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Formats.Tar
+{
+    // Stream that allows wrapping a super stream and specify the lower and upper limits that can be read from it.
+    // It is meant to be used when the super stream is seekable.
+    // Does not support writing.
+    internal sealed class SeekableSubReadStream : SubReadStream
+    {
+        public SeekableSubReadStream(Stream superStream, long startPosition, long maxLength)
```

Comment:
Can we call these something other than "sub" and "super"?  I find the terminology confusing.

Reply:
> Can we call these something other than "sub" and "super"? I find the terminology confusing.

@stephentoub `SuperSubmarineStream`? ;)